app/code/Magento/CatalogInventory/Api/Data/WarehouseRecordInterface.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Magento\CatalogInventory\Api\Data;
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Warehouse record interface
+ *
+ * It is used only to manage stock status
+ * qty of non virtual warehouse inventory
+ *
+ * @api
+ */
+interface WarehouseRecordInterface extends ExtensibleDataInterface
+{
+    /**
+     * Returns identifier of inventory record
+     *
+     * @return int
+     */
+    public function getId();
+
+    /**
+     * Returns identifier of assigned product
+     *
+     * @return int
+     */
+    public function getProductId();
+
+    /**
+     * Returns identifier of assigned inventory
+     *
+     * @return int
+     */
+    public function getInventoryId();
+
+    /**
+     * Returns available quantity of product in the specified warehouse
+     *
+     * @return float
+     */
+    public function getQuantity();
+
+    /**
+     * Sets available quantity of product in the specified warehouse
+     *
+     * @param float $quantity
+     * @return $this
+     */
+    public function setQuantity($quantity);
+
+    /**
+     * Sets a stock status for an item
+     *
+     * @param int $flag
+     * @return $this
+     */
+    public function setIsSalable($flag);
+
+    /**
+     * Sets availability status code for warehouse record
+     *
+     * @param string $code
+     * @return $this
+     */
+    public function setAvailabilityStatusCode($code);
+
+    /**
+     * Returns availability status code for warehouse record
+     *
+     * @return string
+     */
+    public function getAvailabilityStatusCode();
+
+    /**
+     * Returns instance of inventory interface
+     *
+     * Object should be set by inventory record repository
+     *
+     * @return InventoryInterface
+     */
+    public function getInventory();
```

Comment:
Seems this method should be renamed to getWarehouse() which returns an instance of WarehouseInterface as well (according to current logic)

For now we have Warehouse and Inventory set of interfaces very similar to each other. For example, Warehouse interface has all getters provided by Inventory interface. The only difference that Warehouse interfaces are mutable, but inventory - not. 
Why not to create a relation between these kind of interfaces in code? For example. Read-only interfaces of Inventory, and Warehouse interfaces extends (inherit from) them and add mutability logic?

Because for now, if we will decide to add some logic to InventoryInterface in future we will have to add the same methods for WarehouseInterface as well. And this is obscure, because there is no contract in the code.